File: python/send_email.py
import smtplib
from email.message import EmailMessage
import os

def send_email(options):
    # Check for valid email address
    if not options['mail'] or not is_valid_email(options['mail']):
        raise ValueError("Invalid email address")

    # Validate subject
    if not options['subject'] or options['subject'].strip() == "":
        raise ValueError("Subject is required")

    # Validate message
    if not options['message'] or options['message'].strip() == "":
        raise ValueError("Message body is required")

    smtp_host = os.environ.get('SMTP_HOST')
    smtp_port = os.environ.get('SMTP_PORT')
    smtp_user = os.environ.get('SMTP_USER')
    smtp_pass = os.environ.get('SMTP_PASS')
    from_email = os.environ.get('FROM_EMAIL')

    msg = EmailMessage()
    msg['From'] = from_email
    msg['To'] = options['mail']
    msg['Subject'] = options['subject']
    msg.set_content(options['message'])

    with smtplib.SMTP(smtp_host, smtp_port) as server:
        server.login(smtp_user, smtp_pass)
        server.send_message(msg)

def is_valid_email(email):
    # Regular expression to validate email format
    email_regex = r'^[^\s@]+@[^\s@]+\.[^\s@]+$'
    return bool(re.match(email_regex, email))

File: typescript/sendEmail.ts
import nodemailer from "nodemailer";

/**
 * Sends an email using Nodemailer.
 * @param {Object} options - Email options.
 * @param {string} options.mail - Recipient's email address.
 * @param {string} options.subject - Subject of the email.
 * @param {string} options.message - Body of the email.
 */
const sendEmail = async (options: { mail: string; subject: string; message: string }) => {
  // Check for valid email address
  if (!options.mail || !isValidEmail(options.mail)) {
    throw new Error("Invalid email address");
  }

  // validate subject
  if (!options.subject || options.subject.trim() == "") {
    throw new Error("Subject is Required");
  }

  // validate message
  if (!options.message || options.message.trim() == "") {
    throw new Error("Message body is required");
  }

  // Create a transporter with SMTP configuration
  const transporter = nodemailer.createTransport({
    host: process.env.SMTP_HOST, // SMTP server hostname
    port: process.env.SMTP_PORT, // SMTP server port

    // SMTP authentication credentials
    auth: {
      user: process.env.SMTP_USER, // SMTP username
      pass: process.env.SMTP_PASS, // SMTP password
    },
  });

  // Create email message object
  const message = {
    from: process.env.FROM_EMAIL, // Sender's email address
    to: options.mail, // Recipient's email address
    subject: options.subject, // Subject of the email
    text: options.message, // Body of the email
  };

  // Send the email using the transporter
  await transporter.sendMail(message);
};

/**
 * Validate email address format.
 * @param {string} email - Email address to validate.
 * @returns {boolean} - True if email is valid, otherwise false.
 */
const isValidEmail = (email: string): boolean => {
  // Regular expression to validate email format
  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  return emailRegex.test(email);
};

export default sendEmail;

// Author: Abhishek Kumar
// GitHub: https://github.com/abhiraj-ku/sendEmail
